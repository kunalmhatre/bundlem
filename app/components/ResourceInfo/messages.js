/*
 * ResourceInfo Messages
 *
 * This contains all the text for the ResourceInfo component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.ResourceInfo';

export default defineMessages({
  'actionWEB PAGE': {
    id: `${scope}.actionWEB PAGE`,
    defaultMessage: 'VISIT',
  },
  actionVIDEO: {
    id: `${scope}.actionVIDEO`,
    defaultMessage: 'WATCH',
  },
  actionDOCUMENT: {
    id: `${scope}.actionDOCUMENT`,
    defaultMessage: 'VIEW',
  },
  actionIMAGE: {
    id: `${scope}.actionIMAGE`,
    defaultMessage: 'VIEW',
  },
  actionOTHER: {
    id: `${scope}.actionOTHER`,
    defaultMessage: 'VIEW',
  },
  'WEB PAGE': {
    id: `${scope}.WEB PAGE`,
    defaultMessage: 'WEB PAGE',
  },
  VIDEO: {
    id: `${scope}.VIDEO`,
    defaultMessage: 'VIDEO',
  },
  DOCUMENT: {
    id: `${scope}.DOCUMENT`,
    defaultMessage: 'DOCUMENT',
  },
  IMAGE: {
    id: `${scope}.IMAGE`,
    defaultMessage: 'IMAGE',
  },
  OTHER: {
    id: `${scope}.OTHER`,
    defaultMessage: 'OTHER',
  },
});
