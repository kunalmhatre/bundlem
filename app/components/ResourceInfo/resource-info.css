.resource-info {
  font-weight: 700;
  border-bottom: 1px solid;
  padding: 1.5rem 1rem 1.5rem 0;
}

.resource-info-title {
  padding-bottom: 0.5rem;
  font-size: 1.3rem;
}

.resource-info-notes {
  padding: 0.5rem 0 0.5rem 0;
  opacity: 0.7;
}

.resource-info-action {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding-top: 0.5rem;
}

@media only screen and (max-width: 767px) {
  .resource-info-title {
    font-size: 1.04rem;
  }
  .resource-info-notes {
    font-size: 0.8rem;
  }
  .resource-info-action-label {
    font-size: 0.8rem;
  }
}