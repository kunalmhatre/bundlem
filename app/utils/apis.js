/**
 * REST APIs
 */

export const domain = 'https://bundlem.herokuapp.com';

export const verifyTokenAPI = `${domain}/verify`;
export const publishBundleAPI = `${domain}/create`;
export const getBundleAPI = `${domain}/bundle`;
