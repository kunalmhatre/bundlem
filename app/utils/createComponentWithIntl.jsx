/**
 * Renders component with react-intl support for snapshot testing
 * Source: https://github.com/yahoo/react-intl/wiki/Testing-with-React-Intl#helper-function-2
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';

const createComponentWithIntl = (children, props = { locale: 'en' }) => renderer.create(
  <IntlProvider {...props}>
    {children}
  </IntlProvider>,
);

export default createComponentWithIntl;
