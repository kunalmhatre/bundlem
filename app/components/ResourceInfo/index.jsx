import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import Button from '../Button';
import messages from './messages';
import './resource-info.css';

const propTypes = {
  resourceType: PropTypes.oneOf([
    'WEB PAGE',
    'VIDEO',
    'DOCUMENT',
    'IMAGE',
    'OTHER',
  ]),
  resourceLink: PropTypes.string,
  resourceCount: PropTypes.number,
  resourceTitle: PropTypes.string,
  resourceNotes: PropTypes.string,
  onLoad: PropTypes.func,
};

const defaultProps = {
  resourceType: 'OTHER',
  resourceLink: '#',
  resourceCount: 0,
  resourceTitle: null,
  resourceNotes: null,
  onLoad: () => null,
};

/**
 * ResourceInfo component
 * @param {string} resourceType - Type of the resource
 * @param {string} resourceLink - Link of the resource
 * @param {number} resourceCount - Location of the resource (index + 1) from the list of resources
 * @param {string} resourceTitle - Title of the resource
 * @param {string} resourceNotes - Notes of the resource
 * @param {function} onLoad - On load handler function
 */
function ResourceInfo({
  resourceType,
  resourceLink,
  resourceCount,
  resourceTitle,
  resourceNotes,
  onLoad,
}) {
  return (
    <div
      className="resource-info theme-blue"
      onLoad={onLoad}
    >
      <div className="resource-info-title">
        {`${resourceCount}. `}
        {resourceTitle || resourceType}
      </div>
      {
        resourceNotes ? (
          <div className="resource-info-notes">{resourceNotes}</div>
        ) : null
      }
      <div className="resource-info-action">
        <Button
          text={messages[`action${resourceType}`]}
          onClick={() => { window.open(resourceLink); }}
        />
        <span className="resource-info-action-label">
          <FormattedMessage {...messages[resourceType]} />
        </span>
      </div>
    </div>
  );
}

ResourceInfo.propTypes = propTypes;
ResourceInfo.defaultProps = defaultProps;

export default ResourceInfo;
