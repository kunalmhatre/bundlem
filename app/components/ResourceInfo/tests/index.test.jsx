import React from 'react';

import createComponentWithIntl from '../../../utils/createComponentWithIntl';
import { mountWithIntl } from '../../../utils/reactIntlHelperFunction';
import ResourceInfo from '../index';

describe('<ResourceInfo />', () => {
  it('renders old snapshot', () => {
    expect(
      createComponentWithIntl(
        <ResourceInfo />,
      ).toJSON(),
    ).toMatchSnapshot();
  });

  it('shows resource notes when provided with one', () => {
    const testNotes = 'testNotes';

    expect(
      mountWithIntl(
        <ResourceInfo resourceNotes={testNotes} />,
      ).find('div.resource-info-notes')
        .text(),
    ).toBe(testNotes);
  });

  it('shows resource title when provided with one', () => {
    const testCount = 1;
    const testTitle = 'testTitle';

    expect(
      mountWithIntl(
        <ResourceInfo
          resourceCount={testCount}
          resourceTitle={testTitle}
        />,
      ).find('div.resource-info-title')
        .text(),
    ).toBe(`${testCount}. ${testTitle}`);
  });

  it('resource action is functional', () => {
    const testLink = 'https://google.com';
    window.open = jest.fn(resourceLink => resourceLink);

    mountWithIntl(
      <ResourceInfo resourceLink={testLink} />,
    ).find('div.resource-info-action button')
      .simulate('click');

    expect(window.open).toHaveReturnedWith(testLink);
  });

  it('calls default onLoad handler', () => {
    mountWithIntl(
      <ResourceInfo />,
    ).find('div.resource-info')
      .simulate('load');
  });
});
