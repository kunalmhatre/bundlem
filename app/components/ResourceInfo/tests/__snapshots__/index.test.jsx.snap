// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResourceInfo /> renders old snapshot 1`] = `
<div
  className="resource-info theme-blue"
  onLoad={[Function]}
>
  <div
    className="resource-info-title"
  >
    0. 
    OTHER
  </div>
  <div
    className="resource-info-action"
  >
    <button
      className="button theme-blue-inverse"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span>
        VIEW
      </span>
      
    </button>
    <span
      className="resource-info-action-label"
    >
      <span>
        OTHER
      </span>
    </span>
  </div>
</div>
`;
